OpenTerm/Controller/ViewController.swift

```diff
@@ -339,6 +357,261 @@ extension ViewController: UIDocumentPickerDelegate {
+    /// such that the settings view controller can access is as well.
+    static let bookmarkDirectory = ".bookmarks"
+    
+    
+    /// Gets the URLs of the bookmarks that were previously saved.
+    ///
+    /// - Returns: The URLs of the saved bookmarks. If something fails, the returned array will be empty.
+    func savedBookmarkURLs() -> [URL] {
+        /**
+         *  Get the document directory of the app.
+         */
+        if let dir = FileManager.default.urls(for: .documentDirectory,
+                                              in: .userDomainMask).first {
+            
+            /**
+             *  Get the directory where the bookmarks are saved.
+             */
+            let bookmarkDirectoryURL = dir.appendingPathComponent(ViewController.bookmarkDirectory,
+                                                                  isDirectory: true)
+            
+            /**
+             *  Create the bookmarks directory (if it doesn't exist)
+             */
+            do {
+                try FileManager.default.createDirectory(at: bookmarkDirectoryURL,
+                                                        withIntermediateDirectories: true,
+                                                        attributes: nil)
+                
+                /**
+                 *  Get all files that are in the bookmarks directory.
+                 */
+                let bookmarkFiles = try FileManager.default.contentsOfDirectory(atPath: bookmarkDirectoryURL.path)
+                
+                /**
+                 *  The array that will hold the obtained URLs. This will be returned eventually.
+                 */
+                var bookmarkURLs = [URL]()
+                
+                /**
+                 *  Iterate all bookmark filenames.
+                 */
+                for bookmarkFileName in bookmarkFiles {
+                    
+                    /**
+                     *  Get the url of the current bookmark file.
+                     */
+                    let bookmarkDataURL = bookmarkDirectoryURL.appendingPathComponent(bookmarkFileName)
+                    
+                    /**
+                     *  We try to load the bookmark from the file.
+                     */
+                    let loadedBookmark = try URL.bookmarkData(withContentsOf: bookmarkDataURL)
+                    
+                    /**
+                     *  This variable will indicate whether the bookmark is stale.
+                     *  If the bookmark is stale we create a new bookmark with the
+                     *  obtained URL and save the new one instead of the old one.
+                     */
+                    var isStale = true
+                    
+                    /**
+                     *  Try to obtain the URL from the bookmark.
+                     */
+                    if let loadedBookmarkURL = try URL(resolvingBookmarkData: loadedBookmark, bookmarkDataIsStale: &isStale) {
```

Comment:
If one bookmark is unable to resolve, an error is thrown and all parsing will stop. Why not delete the file and continue parsing instead?